channel/src/main/java/com/synopsys/integration/alert/channel/jira2/common/delegate/JiraIssueTransitioner.java

```diff
@@ -0,0 +1,133 @@
+package com.synopsys.integration.alert.channel.jira2.common.delegate;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.alert.channel.api.issue.search.ExistingIssueDetails;
+import com.synopsys.integration.alert.channel.api.issue.send.IssueTrackerIssueCommenter;
+import com.synopsys.integration.alert.channel.api.issue.send.IssueTrackerIssueResponseCreator;
+import com.synopsys.integration.alert.channel.api.issue.send.IssueTrackerIssueTransitioner;
+import com.synopsys.integration.alert.common.channel.issuetracker.enumeration.IssueOperation;
+import com.synopsys.integration.alert.common.channel.issuetracker.exception.IssueMissingTransitionException;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jira.common.model.components.IdComponent;
+import com.synopsys.integration.jira.common.model.components.StatusCategory;
+import com.synopsys.integration.jira.common.model.components.StatusDetailsComponent;
+import com.synopsys.integration.jira.common.model.components.TransitionComponent;
+import com.synopsys.integration.jira.common.model.request.IssueRequestModel;
+import com.synopsys.integration.jira.common.model.response.TransitionsResponseModel;
+import com.synopsys.integration.jira.common.server.builder.IssueRequestModelFieldsBuilder;
+
+public abstract class JiraIssueTransitioner extends IssueTrackerIssueTransitioner<String> {
+    public static final String TODO_STATUS_CATEGORY_KEY = "new";
+    public static final String DONE_STATUS_CATEGORY_KEY = "done";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    protected JiraIssueTransitioner(
+        IssueTrackerIssueCommenter<String> commenter,
+        IssueTrackerIssueResponseCreator issueResponseCreator
+    ) {
+        super(commenter, issueResponseCreator);
+    }
+
+    protected abstract String extractReopenTransitionName();
+
+    protected abstract String extractResolveTransitionName();
+
+    protected abstract StatusDetailsComponent getIssueStatus(String issueKey) throws IntegrationException;
+
+    protected abstract TransitionsResponseModel getIssueTransitions(String issueKey) throws IntegrationException;
+
+    protected abstract void transitionIssue(IssueRequestModel issueRequestModel) throws IntegrationException;
+
+    @Override
+    protected Optional<String> retrieveJobTransitionName(IssueOperation transitionType) {
```

Comment:
This can probably become `final` along with most of the implemented methods in this class.